packages/block-library/src/search/edit.js

```diff
@@ -408,15 +410,34 @@ export default function SearchEdit( {
 			</BlockControls>
 
 			<InspectorControls>
-				<PanelBody title={ __( 'Settings' ) }>
-					<VStack
-						className="wp-block-search__inspector-controls"
-						spacing={ 4 }
+				<ToolsPanel
+					label={ __( 'Settings' ) }
+					resetAll={ () => {
+						setAttributes( {
+							width: undefined,
+							widthUnit: undefined,
+						} );
+					} }
+					dropdownMenuProps={ dropdownMenuProps }
+				>
+					<ToolsPanelItem
+						hasValue={ () =>
+							PERCENTAGE_WIDTHS.includes( width ) &&
+							widthUnit === '%'
+						}
+						label={ __( 'Width' ) }
+						resetAll={ () => {
+							setAttributes( {
+								width: PC_WIDTH_DEFAULT, // Reset to default value
+								widthUnit: undefined,
+							} );
+						} }
```

Comment:
```suggestion
						onDeselect={ () => {
							setAttributes( {
								width: undefined,
								widthUnit: undefined,
							} );
						} }
```

- The `ToolsPanelItem` component does not have a `resetAll` prop. Instead, we use `onDeselect` to reset attributes.
- Resetting means "return that attribute value to its default value." The default for both attributes is `undefined`.